File: first/java/Nighttime.java

public class Nighttime extends Extras {

    private final float nighttime = 1.75f;
    private final int factor;

    public Nighttime(int extraHours) {
        super(extraHours);
        this.factor = extraHours;
    }

    @Override
    public float extraPay(int valueHour) {
        return (valueHour * nighttime) * factor;
    }
}

File: first/php/library/nighttime.php
<?php
class Nighttime extends Extras {
    
    private float $nighttime = 1.75;
    private int $factor;

    public function __construct(int $extraHours) {
        parent::__construct($extraHours);
        $this->factor = $extraHours;
    }

    public function extraPay(int $valueHour): float {
        return ($valueHour * $this->nighttime) * $this->factor;
    }
}
?>